src/provider/provider_os_memory.c

```diff
@@ -347,8 +408,20 @@ static int os_mmap_aligned(void *hint_addr, size_t length, size_t alignment,
         extended_length += alignment;
     }
 
-    void *ptr = os_mmap(hint_addr, extended_length, prot);
+    size_t fd_offset = 0;
+
+    if (fd > 0) {
+        fd_offset = *fd_size;
+        *fd_size += extended_length;
```

Comment:
We are updating the output parameter `fd_size`, but `if` block on the next line might return an error. Should we check the error condition and update the output arguments only if the function successes?

Another question, how practical is to reach the `max_fd_size` limit? I understand that it is `SIZE_MAX` now, but what if workflow causes continuos malloc/free from memory provider?

Reply:
`max_fd_size = max(off_t) = 9223372036854775807 = 2^63-1 = 8388608 TB - 1 = 8192 PB - 1` is not to be reached practically, I think ...

Updating `fd_size` fixed.
Done.

Reply:
@vinser52 Done

Reply:
We were discussing this with @pbalcer and it's exactly what was done in PMDK and nobody ever complained about it. I know it's an anecdotal argument :), but in reality such a scenario where SIZE_MAX is not enough is *very* rare.

Reply:
Like ldorau says, it's not possible for this variable to overflow in practice.

Reply:
yeah, I have the same feeling, just asked to confirm. The scenario I had in mind was a sequence of malloc/free when total memory consumption might not increase, but on each malloc, OS memory provider does not reuse the memory region that was freed before. The `fd_size` always monotonically increases.